frameworks/bitcoin-move/sources/ord.move

```diff
@@ -619,5 +628,52 @@ module bitcoin_move::ord {
         };
     }
 
+    // ==== Inscription Metaprotocol Validity ==== //
+
+    // Seal the metaprotocol validity for the given inscription_id.
+    #[private_generics(T)]
+    public fun seal_metaprotocol_validity<T>(inscription_id: InscriptionID, is_valid: bool, invalid_reason: Option<String>) : Object<MetaprotocolValidity> {
+        let store_obj_id = object::named_object_id<InscriptionStore>();
+        let store_obj = object::borrow_mut_object_shared<InscriptionStore>(store_obj_id);
+
+        let inscription_object_id = derive_inscription_id(inscription_id);
+        let inscription_obj = object::borrow_mut_object_field<InscriptionStore, Inscription>(store_obj, inscription_object_id);
+
+        let protocol_type = type_info::type_name<T>();
+        let validity = MetaprotocolValidity {
+            protocol_type,
+            is_valid,
+            invalid_reason,
+        };
+
+        let validity_id = MetaprotocolValidityID{};
+        object::add_object_field_with_id<Inscription, MetaprotocolValidityID, MetaprotocolValidity>(inscription_obj, validity_id, validity)
```

Comment:
Why is MetaprotocolValidity an Object field? it should be a normal field?

Reply:
I follow the Open-Closed Principle and try not to modify the previous logic with new features. Fixing the previous logic may cause a large number of previous test cases to fail. But it feels like it's over-designed here, so I'd better change it to a normal field to make the logic clearer.

Reply:
I mean, we should use `object::add_field`, not `object::add_object_field`.

I think changing the Inscription struct not a good idea.